pandas/core/indexes/multi.py

```diff
@@ -3659,6 +3660,47 @@ def _validate_fill_value(self, item):
             raise ValueError("Item must have length equal to number of levels.")
         return item
 
+    def putmask(self, mask, value: MultiIndex) -> MultiIndex:
+        """
+        Return a new MultiIndex of the values set with the mask.
+
+        Parameters
+        ----------
+        mask : array like
+        value : MultiIndex
+            Must either be the same length as self or length one
+
+        Returns
+        -------
+        MultiIndex
+        """
+        mask, noop = validate_putmask(self, mask)
+        if noop:
+            return self.copy()
+
+        if len(mask) == len(value):
+            subset = value[mask].remove_unused_levels()
+        else:
+            subset = value.remove_unused_levels()
+
+        new_levels = []
+        new_codes = []
+
+        for i, (value_level, level, level_codes) in enumerate(
+            zip(subset.levels, self.levels, self.codes)
+        ):
+            new_elements = value_level.difference(level)
+            new_level = level.append(new_elements)
```

Comment:
Can the two lines above be replaced with `new_level = level.union(value_level)`?

Reply:
Good point, was thinking about duplicates when implementing this, but levels are unique…